### PR TITLE
Use JsonpMainTemplatePlugin.hooks instead of MainTemplate#hooks

### PR DESCRIPTION
**Pull request by ooflorent**
Refactor `JsonpMainTemplatePlugin` plugin to change how hooks are installed into `MainTemplate`.
cc @jantimon

**What kind of change does this PR introduce?**

refactoring

**Did you add tests for your changes?**

no

**Does this PR introduce a breaking change?**

yes

**What needs to be documented once your changes are merged?**

Maybe `JsonpMainTemplatePlugin.hooks`.

### CODE REVIEW

**Comment by jantimon** [comment]
@sokra @ooflorent I really like that approach and will also use it for the html-webpack-plugin.  
Just one thing - some Plugins might provide a hooks object but others will require a context argument.

Imho we should prefix get functions e.g.: `getHooks` to be more consistent.

`JsonpMainTemplatePlugin.getHooks(`   
vs  
`compilation.hooks`

File: lib/web/JsonpMainTemplatePlugin.js [hunk 1]
```diff
@@ -7,7 +7,22 @@
 const { SyncWaterfallHook } = require("tapable");
 const Template = require("../Template");
 
+const mainTemplateHooksMap = new WeakMap();
+
 class JsonpMainTemplatePlugin {
+	static hooks(mainTemplate) {
```
Comment (sokra) [inline lib/web/JsonpMainTemplatePlugin.js]: `hooks` -> `getHooks`

Comment (ooflorent) [reply lib/web/JsonpMainTemplatePlugin.js]: Alright. I’ll do it in the other PR also

**Comment by sokra** [comment]
Should we add a `instanceof MainTemplate` check? It's very easy to do it wrong, i. e. `JsonpMainTemplatePlugin.getHooks(compilation)`. This doesn't throw, but also don't work as expected. A `instanceof MainTemplate` could help by throwing an error.

**Comment by sokra** [comment]
Thanks

**Pull request merged**